cmd/lsif-server/main.go

```diff
@@ -1,45 +1,103 @@
 package main
 
 import (
+	"fmt"
+	"io/ioutil"
 	"log"
 	"os"
+	"path"
 	"strconv"
 	"strings"
 
+	"github.com/sourcegraph/sourcegraph/internal/env"
 	"github.com/sourcegraph/sourcegraph/internal/goreman"
 )
 
+var (
+	servers = env.Get("LSIF_NUM_SERVERS", "1", "the number of server instances to run (defaults to one)")
+	workers = env.Get("LSIF_NUM_WORKERS", "1", "the number of worker instances to run (defaults to one)")
+
+	// Set in docker image
+	prometheusStorageDir       = os.Getenv("PROMETHEUS_STORAGE_DIR")
+	prometheusConfigurationDir = os.Getenv("PROMETHEUS_CONFIGURATION_DIR")
+)
+
+const (
+	serverPort = 3186
+	workerPort = 3187
+)
+
 func main() {
-	targets, ok := os.LookupEnv("LSIF_RUN_TARGET")
-	if !ok {
-		targets = "server,worker"
+	numServers, err := strconv.ParseInt(servers, 10, 64)
+	if err != nil || numServers < 0 || numServers > 1 {
+		log.Fatalf("invalid int %q for LSIF_NUM_SERVERS: %s", servers, err)
 	}
 
-	procfile := []string{}
-	for _, target := range strings.Split(targets, ",") {
-		switch target {
-		case "server":
-			procfile = append(procfile, `lsif-server: node /lsif/out/server/server.js`)
-		case "worker":
-			procfile = append(procfile, `lsif-worker: node /lsif/out/worker/worker.js`)
-		default:
-			log.Fatalf("Unknown value '%s' for LSIF_RUN_TARGET (expected 'server' or 'worker')", target)
-		}
-	}
-
-	// Shutdown if any process dies
-	procDiedAction := goreman.Shutdown
-	if ignore, _ := strconv.ParseBool(os.Getenv("IGNORE_PROCESS_DEATH")); ignore {
-		// IGNORE_PROCESS_DEATH is an escape hatch that matches the same behavior
-		// as in sourcegraph/server.
-		procDiedAction = goreman.Ignore
-	}
-
-	err := goreman.Start([]byte(strings.Join(procfile, "\n")), goreman.Options{
+	numWorkers, err := strconv.ParseInt(workers, 10, 64)
+	if err != nil || numWorkers < 0 {
+		log.Fatalf("invalid int %q for LSIF_NUM_WORKERS: %s", workers, err)
+	}
+
+	if err := ioutil.WriteFile(
+		path.Join(prometheusConfigurationDir, "targets.yml"),
+		[]byte(makePrometheusTargets(numServers, numWorkers)),
+		0644,
+	); err != nil {
+		log.Fatal(err.Error())
```

Comment:
```suggestion
		log.Fatalf("Writing prometheus config: %v", err)
```

The error from `WriteFile` will mention the filename already, but it's nice to have context for why we were writing there.